src/logic/relation.lean

```diff
@@ -362,6 +362,13 @@ begin
   { rcases IH with ⟨d, had, hdb⟩, exact ⟨_, had, hdb.tail hbc⟩ }
 end
 
+lemma well_founded {α} {r : α → α → Prop} (h : well_founded r) : well_founded (trans_gen r) :=
```

Comment:
I think this might be better as `_root_.well_founded.trans_gen`, as that enables dot notation on `h`?